src/lib/entities/content-type.ts

```diff
@@ -98,20 +95,39 @@ class EditorInterfaces {
     this._version = version
   }
 
-  update (fieldId: string, widgetId: string) {
-    this._controls.set(fieldId, widgetId)
+  update (fieldId: string, widgetId: string, settings?: APIEditorInterfaceSettings) {
+    let control: APIEditorInterfaceControl = find(this._controls, (c) => {
+      return c.fieldId === fieldId
+    })
+
+    if (!control) {
+      control = {
+        fieldId: fieldId,
+        widgetId: widgetId
+      }
+      control.fieldId = fieldId
+      this._controls.push(control)
+    }
+
+    control.widgetId = widgetId
+
+    if (settings) {
+      forEach(settings, (v: any, k: string) => {
+        control.settings = control.settings || {}
+        control.settings[k] = v
+      })
+    }
   }
 
   toAPI (): object {
-    let result: APIEditorInterface[] = []
-    let widgetId: string
-    let fieldId: string
-    for ([fieldId, widgetId] of this._controls.entries()) {
+    let result: APIEditorInterfaceControl[] = []
+    forEach(this._controls, (c) => {
       result.push({
-        fieldId,
-        widgetId
+        fieldId: c.fieldId,
+        widgetId: c.widgetId,
+        settings: c.settings
```

Comment:
What if there's no `c.settings`? shouldn't the payload remove the property entirely instead of sending `settings: undefined`?